### ISSUE TITLE
Edge case where close button does not work

### ISSUE THREAD

**Issue by meriem-matouk** [ngxpert/hot-toast#21]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

I have this simple repro code, just run it and show the 2 toasts, then attempt to close them, the first will close while the second toast won't close.

```ts
import { ChangeDetectionStrategy, Component, TemplateRef } from '@angular/core';
import { bootstrapApplication } from '@angular/platform-browser';
import 'zone.js';
import { HotToastService, provideHotToastConfig } from '@ngxpert/hot-toast';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';

@Component({
  selector: 'app-root',
  standalone: true,
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
  <button (click)="showToast()">
    Show Toast with Material Symbol Icon
  </button>
  `,
})
export class App {
  name = 'Angular';
  constructor(private toast: HotToastService) {}

  showToast() {
    this.toast.error('eroor.....');
  }
}

bootstrapApplication(App, {
  providers: [
    provideHotToastConfig({
      position: 'top-right',
      stacking: 'vertical',
      visibleToasts: 5,
      autoClose: true,
      dismissible: true,
      style: {
        padding: '16px',
      },
    }), // @ngxpert/hot-toast providers
    provideAnimationsAsync(),
  ],
});


```

## Expected behavior

The toast should close when i click on the close button

## Minimal reproduction of the problem with instructions

Here is a stackblitz repro for the issue: https://stackblitz.com/edit/stackblitz-starters-41c25s?file=src%2Fmain.ts

## What is the motivation / use case for changing the behavior?
improve user experience

## Environment
/
<pre><code>
Angular version: 18.0.0

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 20  
- Platform:  Linux 

Others:

</code></pre>

**Comment by rajanchavda** [comment]
@meriem-matouk  Can you please try with [v3.0.1](https://github.com/ngxpert/hot-toast/releases/tag/v3.0.1) !!!

**Comment by meriem-matouk** [comment]
> @meriem-matouk Can you please try with [v3.0.1](https://github.com/ngxpert/hot-toast/releases/tag/v3.0.1) !!!

the same issue with v3.0.1

**Comment by shhdharmen** [comment]
I checked your stackblitz, and it's working as expected in Mac/Edge. Are you using linux?

https://github.com/user-attachments/assets/33478b1b-d36d-49a9-8086-160ba10f089d

**Comment by meriem-matouk** [comment]
@shhdharmen I'm using chrome 

https://github.com/user-attachments/assets/0adb024d-cbac-475b-aa14-bb79ffc63193

**Comment by rajanchavda** [comment]
Yes, there is still some issue on windows. I am also facing the same.
